### PR TITLE
added GA tracking code [RFR]

### PR DESCRIPTION
**Pull request by mandym-webdev**

### CODE REVIEW

File: client/src/index.html [hunk 1]
```diff
@@ -36,6 +36,16 @@
   <link rel="stylesheet" href="http://cdn.leafletjs.com/leaflet-0.7.3/leaflet.css" />
   <link rel="stylesheet" href="main.css">
   <link rel="stylesheet" href="heritage-icon.css">
+  <script>
+    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
+    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
+    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
+    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
+
+    ga('create', 'UA-65100249-1', 'auto');
```
Comment (seanperkins) [inline client/src/index.html]: Please put this in the build process or config file

File: README.md [hunk 2]
```diff
@@ -1 +1,24 @@
...
+}
+```
+
+To run the app: 
+
+* clone the repo to your local system
+* switch to /client
+* run npm install
+* run gulp
+* start the server using `node server`
```
Comment (lifehackett) [inline README.md:32]: May want to mention switching to `/server` first. Also with the changes to .gitignore they will also need to run `npm install` in `/server` before running `node server`

File: client/gulpfile.js [hunk 3]
```diff
@@ -39,6 +41,14 @@ gulp.task('less', function () {
...
 gulp.task('copy', function () {
   gulp.src('./src/index.html')
+    .pipe(
+      htmlreplace({
+        analytics: {
+          src: config.google_analytics_id,
+          tpl: '<script>(function(i,s,o,g,r,a,m){i["GoogleAnalyticsObject"]=r;i[r]=i[r]||function(){(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)})(window,document,"script","//www.google-analytics.com/analytics.js","ga");ga("create", "%s", "auto");ga("send", "pageview");</script>'
+        }
+      })
+    )
```
Comment (lifehackett) [inline client/gulpfile.js:54]: Nice! Never used this package before so I'm going to assume it works

**Comment by lifehackett** [comment]
@mandym-webdev I don't see the `config.json` file as part of this PR?

**Comment by lifehackett** [comment]
The postinstall script in the ROOT `package.json` file (notice "root" in caps so as not to be confused with the other `package.json` files under `/client` and `/server`) will need to be updated to include `npm install` within server to account for the `.gitignore` changes... There might be a better way to script this, but Heroku builds are a new area for me so not super familiar with whats possible.

**Comment by mandym-webdev** [comment]
the config.json is gitignored because it has our GA key in it. I can 
send it to you directly.  Made the change to the readme.

on the root package.json file, it currently has this:

   "scripts": {
     "postinstall": "cd ./client && npm install && node_modules/.bin/gulp",
     "start": "node ./server/server.js"
   }

it looks like npm install is already there? or should I be looking 
elsewhere?

**Comment by lifehackett** [comment]
Why does it matter if the key is checked into source control?  Won't it be visible on the page in prod anyhow? I think it is just an ID as opposed to a private key.

**Comment by lifehackett** [comment]
The current `npm install` takes places in the `/client` directory.  We also need it in the `/server` directory.

I think this should do it.
`"postinstall": "cd ./server && npm install && cd ../client && npm install && node_modules/.bin/gulp"

**Comment by seanperkins** [comment]
I wanted the key moved out from the repo because it is an open source project and if someone else spins up their own, I do not want them accidentally writing to our analytics.

**Comment by seanperkins** [comment]
I also suggested removing node_modules from repo for the server directory because we don't need to include code that can be installed with npm. Mandy did not remove the directory yet but added it to .gitignore.

**Comment by lifehackett** [comment]
@seanperkins Why not restrict the domain on GA? The current approach will break the build and makes maintenance on the project more difficult.  We could take steps to change the build if you really want it this way, but restricting the domain seems the simplest and most robust solution.

**Comment by seanperkins** [comment]
Google does not restrict which domains can push up to analytics. How is this breaking the build?

**Comment by lifehackett** [comment]
Moving the `/server/node_modules` out makes sense to me.  Good catch on removing them from the repo.  

@mandym-webdev do you know how to do that?  I think this is the command `git rm --cached /server/node_modules`, but you can google to verify.  I'm driving back to Boston shortly but can assist later today if you can't get it to work

**Comment by lifehackett** [comment]
Hmmm, I thought you could filter out by domain, but I don't use GA much.  

There is no `config.json` file so I _think_ that gulp task will barf, but perhaps it has better fault tolerance.  I see the README has been updated to mention this so perhaps it isn't a big deal, just does seem quite right...I'll see if my drive inspires any other ideas. 

We would probably want to create a new gulp task for production though and only include GA in that task so local development doesn't create additional noise in GA.

**Comment by mandym-webdev** [comment]
@lifehackett i have made the updates from your comments so it is RFR again. Thanks!

**Comment by lifehackett** [comment]
The changes look good to me.  I still think we should restrict GA when not in production, either by creating a separate build step or adding logic to the script, but that is up to you guys.

**Comment by mandym-webdev** [comment]
yeah, sean and i just talked about it. i'm going to look at it now to 
see about doing that.

**Pull request merged**